src/sync.js

```diff
@@ -1,31 +1,27 @@
+
 import { observe } from 'mobx';
 
 export const syncHistoryWithStore = (history, store) => {
   // Initialise store
   store.history = history;
 
   // Handle update from history object
-  const handleLocationChange = (location) => {
-    store._updateLocation(location);
-  };
+  const handleLocationChange = store._updateLocation;
 
   const unsubscribeFromHistory = history.listen(handleLocationChange);
   handleLocationChange(history.location);
 
-  const subscribe = (listener) => {
-    const onStoreChange = (change) => {
-      const rawLocation = { ...store.location };
-      listener(rawLocation, history.action);
+  const subscribe = listener => {
+    const onStoreChange = () => {
+      listener({ ...store.location }, history.action);
```

Comment:
I think I prefer the intermediate variable personally, and the negative effects are entirely negligible.